src/components/MainTable/MainTable.tsx

```diff
@@ -1,15 +1,58 @@
 import PropTypes from "prop-types";
 import React, { useEffect, useState } from "react";
+import type { ReactNode } from "react";
+import type { SortDirection } from "types";
 
 import Pagination from "../Pagination";
 import Table from "../Table";
+import type { TableProps } from "../Table";
 import TableRow from "../TableRow";
 import TableHeader from "../TableHeader";
 import TableCell from "../TableCell";
 
 import "./MainTable.scss";
+import { TSFixMe } from "index";
 
-const updateSort = (setSortKey, setSortDirection, sortKey, sortDirection) => {
+type Header = {
+  content?: ReactNode;
+  className?: string;
+  sortKey?: string;
+};
+
+type Column = {
+  children: ReactNode;
+  className: string;
+  expanding: boolean;
+  hidden: boolean;
+};
+
+type Row = {
```

Comment:
Could this type be imported from the TableRow component?

Reply:
They seem to have different types:

```
type Row = {
  columns?: Column[];
  expanded?: boolean;
  expandedContent?: ReactNode;
  sortData?: TSFixMe;
};
```

```
// TableRow.tsx
type Props = {
  children: ReactNode;
} & HTMLAttributes<HTMLTableRowElement>;

```

Is `TableRow` incorrect?

Reply:
Ah yes sorry this one is different.

Reply:
Having said that, I think this should still have `} & HTMLAttributes<HTMLTableRowElement>;` as it spreads additional props.